### ISSUE TITLE
Comparison operator provided via variable expansion triggers "Expected comparison operator"

### ISSUE THREAD

**Issue by jasonkarns** [koalaman/shellcheck#1382]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1073, SC1072
- My shellcheck version (`shellcheck --version` or "online"): 0.5.0
- [x] I read the issue's wiki page, e.g. https://github.com/koalaman/shellcheck/wiki/SC2086
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

is_mac() {
  [ "$(uname -s)" = "Darwin" ] || return 1
  [ $# -eq 0 ] || [ "$(osx_version)" "$@" ]
}

if is_mac -ge 1010; then
  echo "mac greater than 1010"
fi
```

(In use in ruby-build and node-build: https://github.com/rbenv/ruby-build/blob/1d9b8129505582b783386d7567d75d7150da1ef4/bin/ruby-build#L103-L106)

#### Here's what shellcheck currently says:
SC1073: Couldn't parse this test expression. Fix to allow more checks.
SC1072: Expected comparison operator (don't wrap commands in []/[[]]). Fix any mentioned problems and try again.

#### Here's what I wanted or expected to see:

I can understand the parser getting tripped up in this case, because the operator is provided as a parameter to the function itself (and expanded via "$@").

Is there a chance this could be accounted for in the parsing? or perhaps a recommended alternative?

**Comment by koalaman** [comment]
You're right, this is currently a limitation in ShellCheck. You can work around it by using the `test` command, where ShellCheck does not try to inspect the contents:

    ... || test "$(osx_version)" "$@"

 In the semi-near future, ShellCheck's parser may allow continuing after unparseable `[` statements so that this can ignored with a directive. There's some existing bug that I can't easily search up right now.
